### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by olpapina**

### CODE REVIEW

File: cypress/e2e/spec.cy.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
+import { homePage } from "../support/pages/homePage"
+import testData from "../fixtures/testData.json"
+
+describe('Amazon Home page tests', () => {
+
+  beforeEach ('open home page', () => {
+    cy.visit('/')
```
Comment (twikssi) [inline cypress/e2e/spec.cy.js]: put it to the home page like openHopePage(){...}

File: cypress/e2e/spec.cy.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  it('verify actual delivery location', () => {
+    let expectedLocation = testData.impossibleDelivery
+    homePage.getActualDeliveryLocation().should('contain', expectedLocation)
+  })
+
+  // it('verify giftcard button click', () => {
+  //   const giftCardPage = homePage.clickGiftCardButton()
+  //   giftCardPage.getEGiftCard().should('contain', 'eGift Cards') && giftCardPage.getPhysicalGiftCard().should('contain', 'Physical gift cards')
+  // })
```
Comment (twikssi) [inline cypress/e2e/spec.cy.js]: avoid commented code

File: cypress/support/pages/productPage.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        return cy.get(this.deliveryValidationMessage).outerText
+    }
+
+    getDeliveryValidationMessage() {
+        return cy.get(this.deliveryValidationMessage)
+    }
+
+    isAddToCartButton(activeStatus) {
+        if (activeStatus==='inactive') {
+            if (cy.get(this.inactiveAddToCart)) return true
```
Comment (twikssi) [inline cypress/support/pages/productPage.js:21]: it always returns true.
https://docs.cypress.io/api/commands/get#Command-Log 
take a look returning value of cy.get()

File: cypress/support/pages/productPage.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            else return false
+        } else {
+            if (cy.get(this.addToCardButton) && cy.get(this.buyNowButton)) return true
```
Comment (twikssi) [inline cypress/support/pages/productPage.js:24]: it always returns true

File: cypress/support/pages/resultPage.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { ProductPage } from "./productPage"
+
+export class ResultPage {
+    constructor() {
+        this.resultImageItems = '[class="a-link-normal s-no-outline"]'
+        this.resultLinks = '[class="a-size-mini a-spacing-none a-color-base s-line-clamp-4"]'
+    }
+
+    clickResultLink(number) {
+        cy.get(this.resultLinks.get(number)).click()
```
Comment (twikssi) [inline cypress/support/pages/resultPage.js:10]: this.resultLinks.get(number) - ???

resultLinks has string value;
string doesn't have method get()

maybe You meant this
cy.get(this.resultLinks).eq(number).click()

File: cypress/support/pages/resultPage.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+        return new ProductPage
+    }
+
+    getResultImageItem(number) {
+        cy.get(this.resultImageItems.get(number))
```
Comment (twikssi) [inline cypress/support/pages/resultPage.js:15]: the same like line 10

File: cypress/e2e/spec.cy.js (hunk 1 above)
Comment (olpapina) [reply cypress/e2e/spec.cy.js]: fixed

File: cypress/e2e/spec.cy.js (hunk 2 above)
Comment (olpapina) [reply cypress/e2e/spec.cy.js]: commented code is deleted

File: cypress/support/pages/productPage.js (hunk 4 above)
Comment (olpapina) [reply cypress/support/pages/productPage.js:24]: will be changed it in task 3

File: cypress/support/pages/productPage.js (hunk 3 above)
Comment (olpapina) [reply cypress/support/pages/productPage.js:21]: will be changed it in task 3

File: cypress/support/pages/resultPage.js (hunk 5 above)
Comment (olpapina) [reply cypress/support/pages/resultPage.js:10]: maybe, I've added components and try to use it in tests, but I would like to take one of the elements in list of links

**Pull request merged**